nixpkgs/nixpkgs.bzl

```diff
@@ -206,6 +206,104 @@ def invert_repositories(f, *args, **kwargs):
 def _readlink(repository_ctx, path):
     return repository_ctx.path(path).realpath
 
+def _nixpkgs_packages_instantiate_impl(repository_ctx):
+    repository_ctx.file("BUILD", "exports_files([\"nix_attrs.nix\"])")
+
+    nix_instantiate_path = _executable_path(
+        repository_ctx,
+        "nix-instantiate",
+        extra_msg = "See: https://nixos.org/nix/",
+    )
+
```

Comment:
The only factorizable repetition here is the `"{foo} = {bar}"` template, and I think that putting that into an auxiliary function will be more cognitive overhead than repeating it twice (especially since functions must be defined at top-level so it would be defined relatively far away from the only place it's used)